### PR TITLE
Inspect remote images built with Platform API 0.5

### LINKED ISSUE #1034: pack inspect-image fails for remote images built with platform API 0.5
### Summary
The default-process parsing logic for API 0.5 in `pack inspect-image` uses docker client `ImageInspectWithRaw` and therefore only works for daemon images
https://github.com/buildpacks/pack/blob/main/inspect_image.go#L140-L143

---

##### Steps
Given a registry image built with platform API 0.5

1. start a registry
```
docker run -p 5000:5000 registry:2
```
2. In `buildpacks/samples`, build one of the sample apps with `--publish`
```
cd apps/bash-script
pack build localhost:5000/samples/bash --builder cnbs/sample-builder:bionic --publish --network host
```
3. Inspect the image
```
pack inspect-image localhost:5000/samples/bash
```

##### Current behavior
```
> pack inspect-image localhost:5000/samples/bash
ERROR: unable to find image 'localhost:5000/samples/bash' locally or remotely
```

##### Expected behavior
It should print information about the image

---

##### pack info
```
Pack:
  Version:  0.16.0+git-e0f6c50.build-1898
  OS/Arch:  darwin/amd64

Default Lifecycle Version:  0.10.1

Supported Platform APIs:  0.3, 0.4
```

##### docker info
N/A

### PR DESCRIPTION
**Pull request by dwillist**
Signed-off-by: dwillist <dthornton@vmware.com>

## Summary

- Add support for running `pack inspect-image` on remote images. 
- Adds support using the `--bom` flag without a docker daemon.

#### Before
- remote images were not inspected, instead local image info was duplicated.
- errors fetching local image `bom`s would error when running on a system without docker.

#### After
- remote images are inspected!
- remote image `bom` can be inspected on systems without a docker daemon

## Documentation

- Should this change be documented?
    - [ ] Yes, see #___
    - [x] No

## Related

Resolves #1034 
~~Requires buildpacks/imgutil#90 to be merged.~~

### CODE REVIEW

File: acceptance/os/variables_darwin.go [hunk 1]
```diff
@@ -5,4 +5,4 @@ package os
 
 import "regexp"
 
-var PackBinaryExp = regexp.MustCompile(`pack-v\d+.\d+.\d+-macos`)
+var PackBinaryExp = regexp.MustCompile(`pack-v\d+.\d+.\d+-macos\.`)
```
Comment (dfreilich) [inline acceptance/os/variables_darwin.go:8]: nit: Remove and rebase

File: go.mod [hunk 2]
```diff
@@ -45,4 +45,6 @@ require (
 	gopkg.in/yaml.v3 v3.0.0-20200615113413-eeeca48fe776
 )
 
+replace github.com/buildpacks/imgutil => github.com/dwillist/imgutil v0.0.0-20210203233218-00d25b882b8a
+
```
Comment (dfreilich) [inline go.mod:49]: remove, and update imgutil

File: inspect_image.go [hunk 3]
```diff
@@ -4,13 +4,13 @@ import (
...
 
-	"github.com/buildpacks/pack/config"
-
 	"github.com/Masterminds/semver"
 	"github.com/buildpacks/lifecycle"
 	"github.com/buildpacks/lifecycle/launch"
 	"github.com/pkg/errors"
 
+	"github.com/buildpacks/pack/config"
+
```
Comment (dfreilich) [inline inspect_image.go:13]:
```suggestion
```

**Review by dfreilich — CHANGES_REQUESTED** [review]

**Comment by dwillist** [comment]
Recent commits on this branch also address the problems in #1095

File: acceptance/acceptance_test.go [hunk 4]
```diff
@@ -1576,20 +1576,13 @@ func testAcceptance(
...
-							defer h.DockerRmi(dockerCli, repoName)
-
-							t.Log("app is runnable")
-							assertMockAppRunsWithOutput(t,
-								assert,
-								repoName,
-								"Launch Dep Contents",
-								"Cached Dep Contents",
-							)
+							// TODO: remove this condition after pack 0.17.2 is released
```
Comment (dfreilich) [inline acceptance/acceptance_test.go:1579]:
```suggestion
							// TODO: remove this condition after pack 0.17.0 is released
```

**Review by dfreilich — APPROVED** [review]
Fantastic! Just small nit in comment

**Pull request merged**